### DISCUSSION TITLE
Ideas for Project

### DISCUSSION THREAD

**Discussion by ohhsodead** [bucanero/apollo-ps3#26] [Ideas]
It's a little too long to explain in detail what I'd like to be able to do, but I already have database modding tool that allows to browse and install mods to the console. Is there anyway I can use your project as project/homebrew for the mods to be installed to on the console. Is there a special place they can be downloaded to and then picked up from your homebrew. It would mean more mods used by apollo-ps3 from a large databse of games saves we've both created.

Also see my [ModioX](https://github.com/ohhsodead/ModioX) project to get an idea of what I mean.

Please add my discord ohhsodead#2519 if you'd like to speak further about this. Thanks!

**Comment by bucanero** [comment]
hello,

Right now Apollo doesn't support installing any mods or external files; the app can edit/download/patch/modify save-game files, because that was basically the original scope of it.

The app can download and unzip `.zip` files using `http/https` , but again, that's limited to savegames. 
From what I understand from your idea, perhaps that feature could be extended, and with a proper database as input data, the app could download a `gamemod.zip` and unzip it to `/dev_hdd0/game/xxxxxx` and just overwrite files.

If you have standard PS3 save-games files (even if they're hacked/maxed out) the easiest way would be to upload them and include them on the Apollo save database, so there's no need to change the app or source code.

I don't use discord much (I actually try to avoid it), but I saw your message on Twitter so we can follow up there, as I'm still not sure if I've got your idea.

**Comment by ohhsodead** [comment]
I was thinking of adding savegames to the database of mods that I have. I just need to know where they can be installed on the console so that they can be picked up from the apollo-ps3 tool when users are using your tool. All mods on the database each have their own .zip file so either they can be download to console somewhere your tool can extract them, or my tool can extract the files and install the manually. But I here that the console database needs to be rebuilt to be shown in the XMB. I'd really like somehow to integrate your tool with mine so the console has somewhere to manage the gamesaves they're installing to the console.

**Comment by bucanero** [comment]
Savegames on PS3 have ownership information, encryption and checksums, so it's not just copying them to the hard drive and rebuild the database.
Even if your tool unpacks, transfers, copy the save game to `/dev_hdd0/`, and rebuild DB, you'd still need to resign the savegame be loaded properly in the game.

these are the paths scanned by Apollo for save-games:
`/dev_hdd0/home/00000xxx/savedata/`
`/dev_usb000/PS3/SAVEDATA/`
`/dev_usb001/PS3/SAVEDATA/`

Still, I think the easiest way would be to upload the save-games here to the Online Save DB, so Apollo can list, download, and unzip the saves directly.

**Comment by ohhsodead** [comment]
Being able to add the mods from by database to yours is out of scope for my tool, for the moment anyways.

So, if the saves folder are upload to `/dev_hdd0/home/00000xxx/savedata/` or `/dev_usb000/PS3/SAVEDATA/`, then your tool can do we needs to be done and apply them to their games?

**Comment by bucanero** [comment]
the best option is to use the USB folders (if you copy to HDD you'll need to rebuild the database)

after a savegame is copied, if you open Apollo the savegame should be listed, and then it could be resigned/unlocked as needed.

**Comment by ohhsodead** [comment]
I will recommended for users to download your application for resigning/unlock their gamesaves once the've been installed using ModioX tool.

Thanks a lot!

**Comment by Michael228489** [comment]
Make sure that saving the PC version of the game turns into a save for the console version of the game, so that it runs smoothly, this is necessary so that you can safely play the game with saving from some fallout (for example, you passed the game, for all sorts of reasons, the save on which you passed the game disappeared, you don't want to go through, download what you need, and the console doesn't see it, you need it to see this save, install it and you would play calmly), the biggest problem here is that games have a variety of save formats for computers (for example, fallout 3 and nv have it in the form of "FOS"), but in all folders with saves I found a file in bak format.
I don't speak english.
Thanks a lot, if you will heard me.
